projects/transformers/experiments/base.py

```diff
@@ -46,7 +46,8 @@
     do_train=True,
     do_eval=True,
     dataloader_drop_last=True,  # keeps consistent batch size
-    num_train_epochs=3  # is overriden if max_steps is defined
+    num_train_epochs=3,  # is overridden if max_steps is defined
+    ddp_find_unused_parameters=False,  # Should not have any unused parameters
```

Comment:
That is interesting, I just looked into the documentation of torch.nn.parallel.DistributedDataParallel to understand what `find_unused_parameters` is doing. Did we observe run time improvements by adding this argument?

Reply:
Yes, It slowed the forward pass.

it is set to `False` by default in pytorch to avoid traversing the autograd graph during the forward pass.
See https://pytorch.org/docs/stable/notes/ddp.html#internal-design